src/WordPressCodeExtractor.php

```diff
@@ -109,21 +111,44 @@ public static function fromDirectory( $dir, Translations $translations, array $o
 	 * Recursively gets all PHP files within a directory.
 	 *
 	 * @param string $dir A path of a directory.
+	 * @param array $exclude List of files and directories to skip.
 	 *
 	 * @return array File list.
 	 */
-	protected static function getFilesFromDirectory( $dir ) {
+	protected static function getFilesFromDirectory( $dir, array $exclude = [] ) {
 		$filtered_files = [];
 
 		try {
 			$files = new RecursiveIteratorIterator(
-				new RecursiveDirectoryIterator( $dir, RecursiveDirectoryIterator::SKIP_DOTS ),
+				new RecursiveCallbackFilterIterator(
+					new RecursiveDirectoryIterator( $dir, RecursiveDirectoryIterator::SKIP_DOTS ),
+					function ( $file, $key, $iterator ) use ( $exclude ) {
+						/** @var SplFileInfo $file */
+						if ( in_array( $file->getBasename(), $exclude, true ) ) {
+							return false;
+						}
+
+						// Check for more complex paths, e.g. /some/sub/folder.
+						foreach( $exclude as $path_or_file ) {
+							if ( substr( $file->getPathname(), - strlen( '/' . $path_or_file ) ) === '/' . $path_or_file ) {
```

Comment:
You should use `substr_compare` instead as it is optimized towards the comparison:

```
$substring = '/' . $path_or_file;
if ( substr_compare( $file->getPathname(), $substring, - strlen( $substring ) ) === 0 ) { }
```

Reply:
Also, I'm wondering how safe this is when using multibyte strings. Off the top of my head, I can't think of a scenario where the comparison would fail to produce the correct result, but I'm not 100% sure. Do you think this might be an issue?

Reply:
Good call.

I can't think of a scenario either _right now_, but you never know :-)

We could use `mb_ereg` in this case:

```php
if ( false !== mb_ereg( preg_quote( '/' . $path_or_file ) . '$', $file->getPathname() ) ) {
	return false;
}
```

Feels more readable anyway.